src/store/state/state-model.ts

```diff
@@ -30,7 +31,8 @@ export const stateModel: StateModelIF = {
       fileNumber: '',
       hasPlanOfArrangement: false
     },
-    validComponents: {
+    flagsCompanyInfo: {
+      // NB: this must be in same order as ComponentsCompanyInfo enum!
```

Comment:
IMHO this is fragile but I didn't have time to fix it. The problem is that properties in an object are not ordered... but we need them ordered to validate them (as per the validation method in common mixin).

It turns out that the order HERE is the order they're enumerated in by `keys()`, but I doubt this is guaranteed.

A better solution would implement a flags "object" that maintains its property order. In fact, we could probably create an array that is a combination of this object and the component IDs enum, which would also automatically keep everything in sync and in order.

Reply:
Yes, I started typing it down about it but then I realized you mentioned it here. I agree.

Reply:
I just think we should find a dynamic solution (maybe getting part of the id instead of exact match) because there are some dynamic components (like ShareStructure).

Reply:
Yes, I could find a solution... but not in this ticket 😁

Reply:
I bet one of us will end up fixing this as part of another ticket (freebie feature) :)